### PR TITLE
Fix(config_flow): Normalize cleared lock select

### PR DESCRIPTION
**Pull request by tykeal**
## Summary

The HA frontend's "X" clear button on select dropdowns sends `None` or
an empty string instead of a value from the allowed list. `vol.In`
rejects these, causing HA to silently re-show the form with old values
— the lock disconnect never actually saves.

## Changes

- Add `_normalize_lock_entry()` in `config_flow.py` that converts
  `None`, empty strings, and whitespace-only strings to the `"(none)"`
  sentinel value
- Wrap the `CONF_LOCK_ENTRY` schema validator with
  `vol.All(_normalize_lock_entry, vol.In(...))` so normalization runs
  before list membership validation
- Add parametrized unit tests for the normalizer function
- Add unit tests verifying that `None` and `""` lock entry values
  pass through the options flow and store `CONF_LOCK_ENTRY = None`

### CODE REVIEW

File: custom_components/rental_control/config_flow.py [hunk 1]
```diff
@@ -133,6 +133,18 @@ async def async_step_init(
...
+def _normalize_lock_entry(value: Any) -> str:
+    """Normalize cleared lock entry values to '(none)'.
+
+    The HA frontend sends None or empty string when the user
+    clears a select field via the X button. Map those to the
+    explicit '(none)' sentinel so vol.In validation passes.
+    """
+    if value is None or (isinstance(value, str) and not value.strip()):
+        return "(none)"
+    return str(value)
```
[the remark this answers is not in the document]
Comment (tykeal) [reply custom_components/rental_control/config_flow.py:145]: Resolved — removed incorrect `Closes #409` reference. This bug was reported in conversation, not via an issue.

File: tests/unit/test_config_flow.py [hunk 2]
```diff
@@ -1147,3 +1149,197 @@ async def test_honor_event_times_toggle_persists(hass: HomeAssistant) -> None:
...
+)
+def test_normalize_lock_entry(input_val: str | None, expected: str) -> None:
+    """Test _normalize_lock_entry maps empty values to (none).
+
+    Verifies that:
+    - None, empty string, and whitespace map to '(none)'
+    - Valid lock names pass through unchanged
+    - The explicit '(none)' sentinel passes through
+    """
+    assert _normalize_lock_entry(input_val) == expected
```
[the remark this answers is not in the document]
Comment (tykeal) [reply tests/unit/test_config_flow.py:1172]: Updated PR description to say "unit tests" instead of "integration tests". These are indeed unit tests under `tests/unit/`.
